### PR TITLE
Show location post

### PR DESCRIPTION
**Pull request by ArgaAlan**

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Message.java [hunk 1]
```diff
@@ -1,17 +1,15 @@
 /*
  * Copyright 2019 Google Inc.
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
```
Comment (acolwell) [inline src/main/java/com/google/codeu/data/Message.java]: Please revert to original formatting.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -14,6 +14,9 @@
 
 package com.google.codeu.data;
 
+import java.util.ArrayList;
```
Comment (acolwell) [inline src/main/java/com/google/codeu/data/Datastore.java]: Please run "mvn com.coveo:fmt-maven-plugin:format" to remove unrelated changes.

File: src/main/webapp/WEB-INF/category.jsp [hunk 3]
```diff
@@ -81,12 +88,16 @@
...
-              <button type="submit" value="Submit">DELETE</button>
-            </form>
+	            <form id="delete-form" action="/messages" method="POST">
+	              <input type="hidden" name="action" value="delete"/>
+	              <input type="hidden" name="callee" value="/country/<%=countryCode%>/c/<%=category%>"/>
+	              <input type="hidden" name="messageID" value="<%=messages.get(i).getId()%>"/>
+	              <button type="submit" value="Submit">DELETE</button>
+	            </form>
+            <% } %>
+            <% if(!messages.get(i).getLat().isEmpty()){ %>
```
Comment (acolwell) [inline src/main/webapp/WEB-INF/category.jsp]: Use hasALocation() here too?

File: src/main/webapp/WEB-INF/country.jsp [hunk 4]
```diff
@@ -40,6 +40,15 @@
       }
 
     </script>
+    <style>
+      <% for(int i = 0; i < messages.size(); i++) { %>
+      		<%if(!messages.get(i).getLat().isEmpty()){%>
```
Comment (acolwell) [inline src/main/webapp/WEB-INF/country.jsp]: Use hasALocation()  here too?

File: src/main/webapp/WEB-INF/country.jsp [hunk 5]
```diff
@@ -122,9 +131,13 @@
                 <button type="submit" value="Submit">DELETE</button>
               </form>
           <% } %>
-          </div>
+          <% if(!messages.get(i).getLat().isEmpty()){ %>
```
Comment (acolwell) [inline src/main/webapp/WEB-INF/country.jsp]: And here too

File: src/main/webapp/WEB-INF/category.jsp [hunk 6]
```diff
@@ -27,6 +27,13 @@
       #map {
         height: 75px;
       }
+      <% for(int i = 0; i < messages.size(); i++) { %>
+      		<%if(!messages.get(i).getLat().isEmpty()){%>
+      		#map<%=i%>{
```
Comment (acolwell) [inline src/main/webapp/WEB-INF/category.jsp]: You shouldn't need to add this for each id. You should just define a class instead and use that.

Just use something like the following here w/o the loop.
.message_map {
  height: 75px; 
}

and then put the following in all your map divs
  class="message_map"

**Review by acolwell — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/codeu/data/Message.java (hunk 1 above)
Comment (ArgaAlan) [reply src/main/java/com/google/codeu/data/Message.java]: Done

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 2 above)
Comment (ArgaAlan) [reply src/main/java/com/google/codeu/data/Datastore.java]: Done=

File: src/main/webapp/WEB-INF/category.jsp (hunk 3 above)
Comment (ArgaAlan) [reply src/main/webapp/WEB-INF/category.jsp]: Done

File: src/main/webapp/WEB-INF/country.jsp (hunk 4 above)
Comment (ArgaAlan) [reply src/main/webapp/WEB-INF/country.jsp]: Done

File: src/main/webapp/WEB-INF/country.jsp (hunk 5 above)
Comment (ArgaAlan) [reply src/main/webapp/WEB-INF/country.jsp]: Done

File: src/main/webapp/WEB-INF/category.jsp (hunk 6 above)
Comment (ArgaAlan) [reply src/main/webapp/WEB-INF/category.jsp]: Done

File: src/main/webapp/WEB-INF/category.jsp [hunk 7]
```diff
@@ -28,6 +28,11 @@
       #map {
         height: 75px;
       }
+		
+	  .message_map {
+		height: 75px;
```
Comment (acolwell) [inline src/main/webapp/WEB-INF/category.jsp]: How about just putting this in user-page.css with the other message related styles instead of here and below so you only need to specify it in one place? Sorry I didn't notice you were defining this in 2 places before..

File: src/main/webapp/WEB-INF/country.jsp [hunk 8]
```diff
@@ -40,6 +40,11 @@
       }
 
     </script>
+    <style>
```
Comment (acolwell) [inline src/main/webapp/WEB-INF/country.jsp]: Remove once once this is moved to user-page.css.

**Review by acolwell — APPROVED** [review]
Looks good with one minor comment.

**Review by acolwell — APPROVED** [review]
